includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -2365,7 +2368,7 @@ public function update_order_status() {
 				// An exception is thrown if an intent can't be found for the given intent ID.
 				$intent    = $this->payments_api_client->get_intent( $intent_id );
 				$status    = $intent->get_status();
-				$charge_id = $intent->get_charge_id();
+				$charge_id = $intent->get_charge()->get_id();
```

Comment:
Similarly to the above, if applicable, we should handle `null` charges here as well.